epilogue-processor/src/main/java/edu/wpi/first/epilogue/processor/LoggerGenerator.java

```diff
@@ -474,4 +530,83 @@ public Boolean visitIdentifier(IdentifierTree identifier, Void unused) {
         },
         null);
   }
+
+  /**
+   * Creates LogMetadata for dependencies of an element marked with @DependsOn annotations.
+   *
+   * @param element the element that has @DependsOn annotations
+   * @param fieldsToLog the list of fields that will be logged
+   * @param methodsToLog the list of methods that will be logged
+   * @return LogMetadata containing dependency names, or null if no valid dependencies
+   */
+  private String generateLogMetadata(Element element, List<VariableElement> fieldsToLog, List<ExecutableElement> methodsToLog) {
+    return generateLogMetadata(element, fieldsToLog, methodsToLog, true);
+  }
+
+  /**
+   * Creates LogMetadata for dependencies of an element marked with @DependsOn annotations.
+   *
+   * @param element the element that has @DependsOn annotations
+   * @param fieldsToLog the list of fields that will be logged
+   * @param methodsToLog the list of methods that will be logged
+   * @param useStaticField if true, return static field reference; if false, return inline construction
+   * @return LogMetadata construction code or field reference, or null if no valid dependencies
+   */
+  private String generateLogMetadata(Element element, List<VariableElement> fieldsToLog, List<ExecutableElement> methodsToLog, boolean useStaticField) {
```

Comment:
Instead of a single method with very different return values based on a boolean flag (which is hard to remember which is which), would it be better to refactor this into a `collectLogDependencies()` method (returning `dependencies` from the current version of the code) and a `generateLogMetadata()` method (using `collectLogDependencies()` and not using `useStaticField`)?